### PR TITLE
Ensure that range is specified when performing an aggregate function over lines

### PR DESCRIPTION
**Pull request by asellitt**
Currently, we are able to perform an aggregation without specifying a range value (Hash Magic). This is not a case that is gracefully handled, and when done, will result in a rather ungraceful explosion. Lets ensure that range is a required argument so that we can safely aggregate

### CODE REVIEW

**Comment by rabidcarrot** [comment]
Nice removal of some cyclic dependencies here :)

File: spec/double_entry/reporting/aggregate_array_spec.rb [hunk 1]
```diff
@@ -5,11 +5,11 @@ module Reporting
       let(:start) { nil }
       let(:finish) { nil }
       let(:range_type) { 'year' }
-      let(:function) { :sum }
+      let(:function) { 'sum' }
```
Comment (rabidcarrot) [inline spec/double_entry/reporting/aggregate_array_spec.rb]: Why this change?

File: lib/double_entry/reporting/aggregate.rb [hunk 2]
```diff
@@ -2,16 +2,19 @@
...
     class Aggregate
-      attr_reader :function, :account, :code, :range, :options, :filter, :currency
+      attr_reader :function, :account, :code, :range, :filter, :currency
 
-      def initialize(function, account, code, options)
+      def self.formatted_amount(function, account, code, range, options = {})
+        new(function, account, code, range, options).formatted_amount
+      end
+
+      def initialize(function, account, code, range, options = {})
```
Comment (rabidcarrot) [inline lib/double_entry/reporting/aggregate.rb]: Ahhh here it is :)

+1

**Comment by rabidcarrot** [comment]
+1

File: spec/double_entry/reporting/aggregate_array_spec.rb (hunk 1 above)
Comment (asellitt) [inline spec/double_entry/reporting/aggregate_array_spec.rb]: there were repeated cases in the spec where we were casting it to a string. I thought: if we're doing that (everywhere its referenced) why not just make it a string?

**Comment by pablolee** [comment]
Nice! :+1:
